tokens/token_impl_test.go

```diff
@@ -0,0 +1,188 @@
+//go:build integration
+// +build integration
+
+package tokens
+
+import (
+	"math/big"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/blndgs/protocol_registry/pkg"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// Sample data for testing
+const sampleEthData = `{
+	"tokens": [
+		{"token_address": "0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48", "name": "USD Coin", "symbol": "USDC", "decimals": 6}
+	],
+	"protocols": [
+		{"address": "0x87870bca3f3fd6335c3f4ce8392d69350b4fa4e2", "name": "AaveV3", "source": true, "destination": true, "tokens": ["0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48"]}
+	]
+}`
+
+const sampleBscData = `{
+	"tokens": [
+		{"token_address": "0x8AC76a51cc950d9822D68b83fE1Ad97B32Cd580d", "name": "USD Coin", "symbol": "USDC", "decimals": 18}
+	],
+	"protocols": [
+		{"address": "0x6807dc923806fE8Fd134338EABCA509979a7e0cB", "name": "AaveV3", "source": true, "destination": true, "tokens": ["0x8AC76a51cc950d9822D68b83fE1Ad97B32Cd580d"]}
+	]
+}`
+
+func TestNewJSONTokenRegistry(t *testing.T) {
+	// Setup: Create temporary JSON files for testing
+	tmpDir := t.TempDir()
+	createTempJSONFile(t, tmpDir, "1.json", sampleEthData)
+	createTempJSONFile(t, tmpDir, "56.json", sampleBscData)
+
+	// Change working directory to temp directory
+	oldWd, _ := os.Getwd()
+	err := os.Chdir(tmpDir)
+	require.NoError(t, err)
+	defer os.Chdir(oldWd)
+
+	// Test
+	registry, err := NewJSONTokenRegistry()
+	require.NoError(t, err)
+	assert.NotNil(t, registry)
+	assert.Len(t, registry.data, 2)
+}
```

Comment:
can you just add the file to `testdata/` folder and use that instead of this?

Reply:
it will b confusion as we already have 1.json and 56.json the function `createTempJSONFile` treats `sampleEthData` and `sampleBscData` as a file